hippod/mime_data_db.py

```diff
@@ -147,9 +157,12 @@ def save_object_item_data(app, data):
         msg = "a data section is required at least: {}".format(str(data))
         raise ApiError(msg)
 
-
     # ok, data stuff
     sha = hippod.hasher.hash_data(data['data'])
+    if data['mime-type'].startswith('x-snippet'):
+        hippod.snippet_db.register_snippet(app, data, sha, source_path, source_type)
```

Comment:
here a validity check should be added. Only python and only mathplotlib allowed, you know. if not raise an error, "snipped executer {} not supported, only python with mathplotlib for now"